File: src/test/java/edu/metrostate/trackingsystem/domain/models/PriceTest.java
package edu.metrostate.trackingsystem.domain.models;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;
class PriceTest {

    @Test
    public void createPrice() {
        var price = new Price();
        assertNotNull(price);
    }

    @Test
    void getPrice() {
        var price = new Price(500, "dollars");
        assertEquals(500, price.getPrice());
    }

    @Test
    void setPrice() {
        var price = new Price(0, "dollars");
        assertEquals(0, price.getPrice());
        price.setPrice(5000);
        assertEquals(5000, price.getPrice());
    }

    @Test
    void getCurrency() {
        var price = new Price(0, "dollars");
        assertEquals("dollars", price.getCurrency());
    }

    @Test
    void setCurrency() {
        var price = new Price(0, "dollars");
        assertEquals("dollars", price.getCurrency());
        price.setCurrency("pounds");
        assertEquals("pounds", price.getCurrency());
    }

    @Test
    void testToString() {
        assertPrice("dollars", "$ 50.00 dollars");
        assertPrice("pounds", "£ 50.00 pounds");
        assertPrice("euros", "€ 50.00 euros");
        assertPrice("yen", "¥ 50.00 yen");
        assertPrice("rupees", "₹ 50.00 rupees");
        assertPrice("whatever", "50.00 whatever");
    }

    void assertPrice(String currency, String expected) {
        var price = new Price(50, currency);
        assertEquals(expected, price.toString());
    }
}

File: src/test/kotlin/edu/metrostate/dealership/domain/models/PriceTest.kt
package edu.metrostate.dealership.domain.models

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

internal class PriceTest {

    @Test
    fun createPrice() {
        val price = Price(500.0, "dollars")
        assertNotNull(price)
    }

    @Test
    fun getPrice() {
        val price = Price(500.0, "dollars")
        assertEquals(500.0, price.price)
    }

    @Test
    fun setPrice() {
        val price = Price(0.0, "dollars")
        assertEquals(0.0, price.price)
        price.price = 5000.0
        assertEquals(5000.0, price.price)
    }

    @Test
    fun getCurrency() {
        val price = Price(0.0, "dollars")
        assertEquals("dollars", price.currency)
    }

    @Test
    fun setCurrency() {
        val price = Price(0.0, "dollars")
        assertEquals("dollars", price.currency)
        price.currency = "pounds"
        assertEquals("pounds", price.currency)
    }

    @Test
    fun testToString() {
        assertPriceToString("dollars", "$50.00 dollars")
        assertPriceToString("pounds", "£50.00 pounds")
        assertPriceToString("euros", "€50.00 euros")
        assertPriceToString("yen", "¥50.00 yen")
        assertPriceToString("rupees", "₹50.00 rupees")
        assertPriceToString("whatever", "50.00 whatever")
    }

    private fun assertPriceToString(currency: String, expected: String) {
        val price = Price(50.0, currency)
        assertEquals(expected, price.toString())
    }
}
